src/Lykke.MatchingEngine.Connector/Services/ClientTcpSocket.cs

```diff
@@ -91,13 +89,20 @@ TService IClientSocketConsumer<TService>.GetConnection()
             return _service;
         }
 
-        // Метод, который пытается сделать соединение с сервером
         private async Task<TcpConnection> Connect()
         {
             _log?.Add("Attempt To Connect:" + _ipEndPoint.Address + ":" + _ipEndPoint.Port);
 
             var tcpClient = new TcpClient { NoDelay = true };
-            await tcpClient.ConnectAsync(_ipEndPoint.Address, _ipEndPoint.Port);
+            try
+            {
+                await tcpClient.ConnectAsync(_ipEndPoint.Address, _ipEndPoint.Port);
+            }
+            catch (Exception)
+            {
+                tcpClient.Dispose();

```

Comment:
Dispose() on finally, not catch?

Reply:
I presume that in case of no exceptions tcpClient is required in no disposed state further as a filed of TcpConnection, right?

Reply:
Yes. @starkmsu is right. Here is the only place where tcpClient can be disposed in case of an exception in   await tcpClient.ConnectAsync(_ipEndPoint.Address, _ipEndPoint.Port);

Reply:
I see, ok